File: scripts/verify_short_reflections.js
const fs = require('fs');
const data = JSON.parse(fs.readFileSync('/Users/anishmoonka/gita-app/data/gita_snippets.json', 'utf8'));

console.log('SHORT REFLECTIONS FOR SNIPPETS 1-20\n');
console.log('='.repeat(80) + '\n');

for (let i = 0; i < 20; i++) {
  const snippet = data.snippets[i];
  console.log(`SNIPPET ${snippet.id}: ${snippet.title}`);
  console.log('-'.repeat(80));
  if (snippet.shortReflection) {
    console.log(snippet.shortReflection);
    console.log(`\nLength: ${snippet.shortReflection.length} characters`);
    const sentences = snippet.shortReflection.split(/[.!?]+/).filter(s => s.trim().length > 0);
    console.log(`Sentences: ${sentences.length}`);
  } else {
    console.log('❌ NO SHORT REFLECTION');
  }
  console.log('\n' + '='.repeat(80) + '\n');
}

console.log('SUMMARY');
console.log('-'.repeat(80));
const count = data.snippets.slice(0, 20).filter(s => s.shortReflection).length;
console.log(`✓ ${count}/20 snippets have shortReflection field`);

File: scripts/verify_short_reflections.py
#!/usr/bin/env python3
"""
Verify and display all shortReflection fields for snippets 1-20
"""

import json

def main():
    input_file = "/Users/anishmoonka/gita-app/data/gita_snippets_hindi.json"

    with open(input_file, 'r', encoding='utf-8') as f:
        data = json.load(f)

    print("=" * 80)
    print("SHORT REFLECTIONS VERIFICATION (Snippets 1-20)")
    print("=" * 80)
    print()

    for snippet in data['snippets'][:20]:
        snippet_id = snippet['id']
        title = snippet['title']
        short_reflection = snippet.get('shortReflection', 'NOT FOUND')

        print(f"Snippet {snippet_id}: {title}")
        print(f"shortReflection: {short_reflection}")
        print()
        print("-" * 80)
        print()

    print("\n✓ All 20 snippets verified successfully!")

if __name__ == '__main__':
    main()
